packages/core/core/src/core-api/routes/validation/utils.ts

```diff
@@ -6,6 +6,39 @@
 
 import * as z from 'zod/v4';
 
+/**
+ * Transforms a Strapi UID into an OpenAPI-compliant component name.
+ *
+ * @param uid - The Strapi UID to transform (e.g., "basic.seo", "api::category.category", "plugin::upload.file")
+ * @returns The OpenAPI-compliant component name (e.g., "BasicSeoEntry", "ApiCategoryCategoryDocument", "PluginUploadFileDocument")
+ */
+export const transformUidToValidOpenApiName = (uid: string): string => {
```

Comment:
Just noticed we're weren't validating that uid is actually valid anywhere in the parent callers, maybe we should add a sanity check for empty string, invalid uid like extra ::, etc and log a warning if we find one?

Reply:
Good idea, I think that's good enough here; this will force us to actually check our inputs before calling